ramalama/model_store.py

```diff
@@ -306,6 +379,52 @@ def new_snapshot(self, model_tag: str, snapshot_hash: str, snapshot_files: list[
 
             os.symlink(blob_relative_path, self.get_snapshot_file_path(snapshot_hash, file.name))
 
+    def _ensure_chat_template(self, model_tag: str, snapshot_hash: str, snapshot_files: list[SnapshotFile]):
+        model_file: SnapshotFile = None
+        for file in snapshot_files:
+            # Give preference to a chat template that has been specified in the file list
+            if file.type == SnapshotFileType.ChatTemplate:
+                return
+            if file.type == SnapshotFileType.Model:
+                model_file = file
```

Comment:
Should break here?  Can there be multiple SnapshotfileType.Model, if yes then we only see the last.

Reply:
The ModelStore would allow multiple models at the moment and huggingface allows that as well (e.g. [mradermacher/SmolLM-135M-GGUF](https://huggingface.co/mradermacher/SmolLM-135M-GGUF/tree/main)). However, this would be invalid for ramalama as input. When the `refs` file is serialized, the current approach is to use the last seen model file - the same applies to the chat template file. So currently, I think its probably worth to add some kind of validaton for this (i.e. only one model and chat template in the list of files) and raise an Exception. WDYT?

Reply:
Yes I think we should throw an error, since RamaLama has no way to know if it chose the right one.

Reply:
Added a validation function, incl. a few unit test cases.